### PR TITLE
[DOC] add TEMPERATURE_MODEL_PARAMETERS to api.rst

### LINKED ISSUE #939: TEMPERATURE_MODEL_PARAMETERS is undocumented
**Is your feature request related to a problem? Please describe.**
`TEMPERATURE_MODEL_PARAMETERS` is mentioned several times in the docs, but it isn't published in the api

**Describe the solution you'd like**
add a docstring after the constant definition in [`temperature.py`](../blob/e526b55365ab0f4c473b40b24ae8a82c7e42f892/pvlib/temperature.py#L9) like this (exact wording tbd):
```python
TEMPERATURE_MODEL_PARAMETERS = {
    'sapm': {
        'open_rack_glass_glass': {'a': -3.47, 'b': -.0594, 'deltaT': 3},
        'close_mount_glass_glass': {'a': -2.98, 'b': -.0471, 'deltaT': 1},
        'open_rack_glass_polymer': {'a': -3.56, 'b': -.0750, 'deltaT': 3},
        'insulated_back_glass_polymer': {'a': -2.81, 'b': -.0455, 'deltaT': 0},
    },
    'pvsyst': {'freestanding': {'u_c': 29.0, 'u_v': 0},
               'insulated': {'u_c': 15.0, 'u_v': 0}}
}
"""Dictionary of temperature parameters organized by model.

There are keys for each model at the top level. Currently there are two models,
``sapm`` for the Sandia Array Performance Model, and ``pvsyst``. Each model has
a dictionary of configurations with a dictionary of the model parameters
associated with it. Retrieve a parameters by indexing the model and
configuration by name.

Example
-------
Retrieve the open rack glass-polymer configuration for SAPM::

    from pvlib.temperature import TEMPERATURE_MODEL_PARAMS
    temp_params = TEMPERATURE_MODEL_PARAMETERS['sapm']['open_rack_glass_polymer']
"""
```

Then somewhere in the docs add an [`autodata` directive](https://www.sphinx-doc.org/es/master/usage/extensions/autodoc.html#directive-autodata) to publish the constant data to the api, like this:
```
Temperature Model Parameters
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.. currentmodule:: pvlib.temperature
.. autodata:: TEMPERATURE_MODEL_PARAMETERS
   :annotation:
```
_note_: remove `:annotation:` to show the value of the dictionary

depending on where it goes, you might have to return the module namespace to whatever it was before this entry using something like this:

    .. currentmodule:: pvlib

to return it to `pvlib` for example if this was in `api.rst`

renders like this:
image

And now it's in the index
image

and can be referenced everywhere in the docs using:

    :data:`~pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS`

Ideas for where to put it?
- in `api.rst` in the subsection called "PV Temperature Models" part of "PV Modeling"
- in `pvsystem.rst` at the end after SAT, in a new section called "temperature"
- huh?

**Describe alternatives you've considered**
- instead of using a dictionary, use a constant class and then add it to the temperature section of `api.rst` after the Faiman model, here's an example:

```python
class _IndexAttrMeta(type):
    """private meta class to index class attributes like a dictionary"""
    def __getitem__(self, item):
        # retrieve items by index like a dictionary
        return getattr(self, str(item).lower())


class TEMPERATURE_MODEL_PARAMETERS(metaclass=_IndexAttrMeta):
    """Dictionary of temperature parameters organized by model.

    There are keys for each model at the top level. Currently there are two models,
    ``sapm`` for the Sandia Array Performance Model, and ``pvsyst``. Each model has
    a dictionary of configurations with a dictionary of the model parameters
    associated with it. Retrieve a parameters by indexing the model and
    configuration by name.

    Example
    -------
    Retrieve the open rack glass-polymer configuration for SAPM::

        from pvlib.temperature import TEMPERATURE_MODEL_PARAMS
        temp_params = TEMPERATURE_MODEL_PARAMETERS['sapm']['open_rack_glass_polymer']
    """
    # NOTE: only Python-2 uses __metaclass__,
    # Python-3 passes this to the class as an argument,
    # see: https://docs.python.org/3/library/2to3.html#2to3fixer-metaclass
    # __metaclass__ = _IndexAttrMeta

    #: Sandia Array Performance Model
    sapm = {
        'open_rack_glass_glass': {'a': -3.47, 'b': -.0594, 'deltaT': 3},
        'close_mount_glass_glass': {'a': -2.98, 'b': -.0471, 'deltaT': 1},
        'open_rack_glass_polymer': {'a': -3.56, 'b': -.0750, 'deltaT': 3},
        'insulated_back_glass_polymer': {'a': -2.81, 'b': -.0455, 'deltaT': 0}}
    #: PVSyst model
    pvsyst = {'freestanding': {'u_c': 29.0, 'u_v': 0},
              'insulated': {'u_c': 15.0, 'u_v': 0}}
```

So this works just like it did before:
```python
In [1]: from pvlib.temperature import TEMPERATURE_MODEL_PARAMETERS

In [2]: TEMPERATURE_MODEL_PARAMETERS['sapm']
Out[2]:
{'open_rack_glass_glass': {'a': -3.47, 'b': -0.0594, 'deltaT': 3},
 'close_mount_glass_glass': {'a': -2.98, 'b': -0.0471, 'deltaT': 1},
 'open_rack_glass_polymer': {'a': -3.56, 'b': -0.075, 'deltaT': 3},
 'insulated_back_glass_polymer': {'a': -2.81, 'b': -0.0455, 'deltaT': 0}}

In [3]: TEMPERATURE_MODEL_PARAMETERS['pvsyst']
Out[3]: {'freestanding': {'u_c': 29.0, 'u_v': 0}, 'insulated': {'u_c': 15.0, 'u_v': 0}}
```

and all of the tests pass
<details>

```
$ pytest pvlib/tests/test_temperature.py
================================================= test session starts =================================================
platform win32 -- Python 3.6.8, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: C:\Users\mikm\Projects, inifile: pytest.ini
plugins: cov-2.8.1, mock-2.0.0, remotedata-0.3.2, rerunfailures-8.0, timeout-1.3.4
collected 14 items

pvlib\tests\test_temperature.py ..............                                                                   [100%]

================================================== warnings summary ===================================================
pvlib\tests\test_temperature.py:5
  C:\Users\mikm\Projects\pvlib-python\pvlib\tests\test_temperature.py:5: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
    from pandas.util.testing import assert_series_equal

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================ 14 passed, 1 warning in 0.07s ============================================
```
</details>

Now just add it to `api.rst`
```
PV temperature models
---------------------

.. autosummary::
   :toctree: generated/

   temperature.sapm_cell
   temperature.sapm_module
   temperature.pvsyst_cell
   temperature.faiman
   temperature.TEMPERATURE_MODEL_PARAMETERS
```
and it will be documented on it's own generated page like other functions
image
image

currently, I can't figure out how to remove the `__init__` method which is not supposed to be documented, but this isn't my first  choice anyway, so if this alternate is the way to go, perhaps someone else can figure out how to remove it?

* another option would be to just make it a class, but this would require changing `TEMPERATURE_MODEL_PARAMETERS` usage everywhere to use attributes instead of items, like `TEMPERATURE_MODEL_PARAMETERS.sapm`. This option is probably the easiest technically, and safest for readability, but requires some grunt work to find all useage and replace indexing with dot notation.

```python
class TEMPERATURE_MODEL_PARAMETERS:
    """Dictionary of temperature parameters organized by model.

    There are keys for each model at the top level. Currently there are two models,
    ``sapm`` for the Sandia Array Performance Model, and ``pvsyst``. Each model has
    a dictionary of configurations with a dictionary of the model parameters
    associated with it. Retrieve a parameters by indexing the model and
    configuration by name.

    Example
    -------
    Retrieve the open rack glass-polymer configuration for SAPM::

        from pvlib.temperature import TEMPERATURE_MODEL_PARAMS
        temp_params = TEMPERATURE_MODEL_PARAMETERS['sapm']['open_rack_glass_polymer']
    """
    #: Sandia Array Performance Model
    sapm = {
        'open_rack_glass_glass': {'a': -3.47, 'b': -.0594, 'deltaT': 3},
        'close_mount_glass_glass': {'a': -2.98, 'b': -.0471, 'deltaT': 1},
        'open_rack_glass_polymer': {'a': -3.56, 'b': -.0750, 'deltaT': 3},
        'insulated_back_glass_polymer': {'a': -2.81, 'b': -.0455, 'deltaT': 0}}
    #: PVSyst model
    pvsyst = {'freestanding': {'u_c': 29.0, 'u_v': 0},
              'insulated': {'u_c': 15.0, 'u_v': 0}}
```

* even a more complicated option would be to figure out how use properties to make the models immutable, but just like `__getitem__`, the property decorator only acts on _instances_ so you would have to figure out how to add it to a metaclass, which is really going too far now:
<details>

```python
class Meta(type):
    @staticmethod
    def bind_geta(a):
        # bind `a` so it doesn't change in the loop
        def geta(self):
            return getattr(self, a)
        # return the bound function `geta`
        return geta
        
    def __getitem__(self, arg):
        return getattr(self, arg)

    def __new__(cls, name, bases, attrs):
        for a in attrs:
            if a.startswith('__'):
                continue
            if not a.startswith('_'):
                # create the property as an attribute of Meta
                setattr(cls, a[1:], property(Meta.bind_geta(a)))
        return super().__new__(cls, name, bases, attrs)


class TEMPERATURE_MODEL_PARAMETERS(metaclass=Meta):
    """Dictionary of temperature parameters organized by model.

    There are keys for each model at the top level. Currently there are two models,
    ``sapm`` for the Sandia Array Performance Model, and ``pvsyst``. Each model has
    a dictionary of configurations with a dictionary of the model parameters
    associated with it. Retrieve a parameters by indexing the model and
    configuration by name.

    Example
    -------
    Retrieve the open rack glass-polymer configuration for SAPM::

        from pvlib.temperature import TEMPERATURE_MODEL_PARAMS
        temp_params = TEMPERATURE_MODEL_PARAMETERS['sapm']['open_rack_glass_polymer']
    """
    _sapm = {
        'open_rack_glass_glass': {'a': -3.47, 'b': -.0594, 'deltaT': 3},
        'close_mount_glass_glass': {'a': -2.98, 'b': -.0471, 'deltaT': 1},
        'open_rack_glass_polymer': {'a': -3.56, 'b': -.0750, 'deltaT': 3},
        'insulated_back_glass_polymer': {'a': -2.81, 'b': -.0455, 'deltaT': 0}}
    _pvsyst = {'freestanding': {'u_c': 29.0, 'u_v': 0},
              'insulated': {'u_c': 15.0, 'u_v': 0}}
```
</details>

**Additional context**
see #938

### PR DESCRIPTION
**Pull request by mikofski**
- [x] Closes #939
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] ~Tests added~ no code, only documentation
 - [ ] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [ ] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

add `TEMPERATURE_MODEL_PARAMETERS` to the documentation

### CODE REVIEW

**Comment by mikofski** [comment]
See if you like what this looks like in the [API reference](https://pvlib-python--1036.org.readthedocs.build/en/1036/api.html#pv-temperature-models)
image
and where it's referenced in the docs, EG: [`pvlib.temperature.sapm_cell`](https://pvlib-python--1036.org.readthedocs.build/en/1036/generated/pvlib.temperature.sapm_cell.html#pvlib.temperature.sapm_cell)
image
and in the [module index](https://pvlib-python--1036.org.readthedocs.build/en/1036/genindex.html#T)
image

**Comment by mikofski** [comment]
also if it's okay to make an exception for stickler, it's a long line in the example of the docstring, I don't think it will look as good wrapped, but maybe?

**Comment by wholmgren** [comment]
Thanks @mikofski! Is it easy to put the whole docstring on its own page and keep only the first line on the rendered `api.rst`? I'm thinking of something like the table entries that surround it. Not a big deal for a single constant but if we add more then a lot of the real estate on the rendered page will be used for constants.

**Comment by mikofski** [comment]
I had troubles with that, b/c the `autosummary` directive only works on classes and functions that have the `__module__` attr - so it adds `TEMPERATURE_MODEL_PARAMS` to the table as a one-liner, but then the link goes to a 404 page not found default page. There may be an option to manually add something to `api.rst`, that's not part of the table, and link it to a new page in the `index.rst` but then it won't be part of the api tree in the left margin What do you think of a page just for constants?

**Comment by wholmgren** [comment]
I was afraid it wasn't going to be easy. I suggest we keep them on `api.rst` until it becomes a problem.

File: pvlib/temperature.py [hunk 1]
```diff
@@ -16,6 +16,21 @@
     'pvsyst': {'freestanding': {'u_c': 29.0, 'u_v': 0},
                'insulated': {'u_c': 15.0, 'u_v': 0}}
 }
+"""Dictionary of temperature parameters organized by model.
+
+There are keys for each model at the top level. Currently there are two models,
+``'sapm'`` for the Sandia Array Performance Model, and ``'pvsyst'``. Each model
+has a dictionary of configurations; a value is itself a dictionary containing model
```
Comment (wholmgren) [inline pvlib/temperature.py:23]: @mikofski can you fix this line length issue? ok to ignore the one in the example (or add # noqa: E501)

**Comment by mikofski** [comment]
Thanks @wholmgren I've fixed the stickler issues, and caught a typo. I think it's good now

**Comment by wholmgren** [comment]
Thanks @mikofski! Test failure is unrelated so merging.

**Pull request merged**